playlist/views.py

```diff
@@ -179,3 +181,32 @@ def post(self, *args, **kwargs):
         keyword = self.request.POST['keyword']
         playlists = Playlist.objects.filter(title__icontains=keyword)
         return render(self.request, self.template_name, {'playlists':playlists})
+
+class AddToPlaylistFromYoutube(LoginRequiredMixin, TemplateView):
+    """ Add to playlist from youtube search
+    """
+    template_name = 'playlist/playlist.html'
+
+    def post(self, *args, **kwargs):
+        playlist = get_object_or_404(
+            Playlist,
+            id=self.request.POST.get('playlist')
+        )
+        form = SongForm(
+            self.request.POST,
+            user=self.request.user,
+            playlist=playlist,
+        )
+        if form.is_valid():
+            form.save()
+            return redirect('playlist', self.request.POST.get('playlist'))
+        # return validation error on playlist
+        songs = Song.objects.filter(playlist=playlist, archive=False)
```

Comment:
i think you don't need to get songs from the playlist. If so, please explain.

Reply:
Updated.